### ISSUE TITLE
Empty webview

### ISSUE THREAD

**Issue by stappersg** [w4/rgit#16]
`rgit` is build and started as
```text
rgit --db-store /var/lib/rgit [::]:49418 /srv/git/
```

With webbrowser I see
![webbrower view](https://github.com/w4/rgit/assets/35434094/42b32ecc-2b99-447b-bc2b-58b3df23b0d0)

I would have expect to see _git repositories_ `foo` and `baz`.

```text
$ ls /srv/git
baz.git  foo.git
$ ls /srv/git/foo.git
branches  config  description  HEAD  hooks  info  objects  refs
$ ls -l /srv/git/foo.git
totaal 32
drwxr-xr-x  2 stappers stappers 4096 17 aug 08:12 branches
-rw-r--r--  1 stappers stappers   99 25 aug 21:19 config
-rw-r--r--  1 stappers stappers   73 17 aug 08:12 description
-rw-r--r--  1 stappers stappers   21 17 aug 12:25 HEAD
drwxr-xr-x  2 stappers stappers 4096 17 aug 08:21 hooks
drwxr-xr-x  2 stappers stappers 4096 19 aug 16:40 info
drwxr-xr-x 61 stappers stappers 4096 17 aug 22:49 objects
drwxr-xr-x  4 stappers stappers 4096 17 aug 08:12 refs
$ cat /srv/git/foo.git/config 
[core]
	repositoryformatversion = 0
	filemode = true
	bare = true
[gitweb]
	owner = Geert Stappers
$ 
```
It seems that there is _db-store content_:
```text
$ ls -l /var/lib/rgit/
totaal 524
drwxr-xr-x 2 root root   4096 25 aug 21:43 blobs
-rw-r--r-- 1 root root     61 25 aug 21:43 conf
-rw-r--r-- 1 root root 524287 25 aug 21:59 db
-rw-r--r-- 1 root root     55 25 aug 21:50 snap.0000000000000092
$
```

What should be done to get beyond empty webview?

**Comment by graskaas** [comment]
> What should be done to get beyond empty webview?

Do you have an initial commit? Initialising a bare repo and then using that doesn't work for me, but cloning a repo with a single commit using `--bare` does. That leads me to think that repos aren't picked up until there is at least one commit.

**Comment by stappersg** [comment]
Only `foo` has content.
`baz` was empty, it never had commits.

I did further tests and got interesting error messages.
(Mostly colleteral of my experimental tests)

```raw
aug 26 19:24:46 hostname rgit[19238]: thread 'main' panicked at 'called
   `Result::unwrap()` on an `Err` value: Io(Os { code: 13, kind:
   PermissionDenied, message: "Permission denied" })', src/main.rs:60:10
aug 26 19:40:26 hostname rgit[19452]: thread '<unnamed>' panicked at
   'called `Result::unwrap()` on an `Err` value: Error { code: -36,
   klass: 7, message: "repository path '/srv/git/baz.git/' is not owned
   by current user" }', src/database/indexer.rs:47:79
aug 26 19:46:24 hostname rgit[19546]: thread '<unnamed>' panicked at
   'called `Result::unwrap()` on an `Err` value: Error { code: -36,
   klass: 7, message: "repository path '/srv/git/baz.git/' is not owned
   by current user" }', src/database/indexer.rs:47:79
aug 26 19:48:28 hostname rgit[19610]: thread '<unnamed>' panicked at
   'called `Result::unwrap()` on an `Err` value: Error { code: -36,
   klass: 7, message: "repository path '/srv/git/baz.git/' is not owned
   by current user" }', src/database/indexer.rs:47:79
```

I did `rm -rf /srv/git/baz.git` to get ride of the repository that triggered errors.

When in `/srv/git/foo.git` doing `git log --oneline` there about eight commits shown.

Sadly still an empty webview.

**Comment by stappersg** [comment]
![beyond empty](https://github.com/w4/rgit/assets/35434094/b05f2997-77f3-4538-9e4c-a29b1dac3574)

I think I have to review the README.

**Comment by stappersg** [comment]
On Sat, Aug 26, 2023 at 04:34:00PM -0700, Geert Stappers wrote:
> I think I have to review the README.

Done.

Now in doubt if "Empty webview" is a reading documentation thingy
or a software feature that can be considered a bug.

Leaving this issue open for collecting additional information.

Do know that it is okay to contact me outside this website.


Groeten
Geert Stappers
-- 
Silence is hard to parse

**Comment by stappersg** [comment]
Probably better worded in [possible bug: bare git without packed-refs file wouldn't show up in the scan result](https://github.com/w4/rgit/issues/50)

**Comment by w4** [comment]
Resolved in https://github.com/w4/rgit/issues/50
